tests/test_connection_manager.py

```diff
@@ -60,37 +71,98 @@ def test_boto_session_with_no_iam_role_and_no_cache(self, mock_Session):
 
         self.connection_manager._boto_session = None
         self.connection_manager.iam_role = None
+        self.connection_manager.profile = None
 
         boto_session = self.connection_manager.boto_session
         assert boto_session.isinstance(mock_Session(
-            region_name="eu-west-1"
+            region_name="eu-west-1",
+            profile=None
         ))
         mock_Session.assert_called_once_with(
-            region_name="eu-west-1"
+            region_name="eu-west-1",
+            profile=None
         )
 
     @patch("sceptre.connection_manager.boto3.session.Session")
-    @patch("sceptre.connection_manager.boto3.client")
-    def test_boto_session_with_iam_role_and_no_cache(
-            self, mock_client, mock_Session
-    ):
-        mock_Session.return_value = sentinel.session
+    def test_boto_session_with_no_iam_role_and_profile(self, mock_Session):
+        mock_Session = MagicMock(name='Session', return_value=sentinel.session)
+        mock_Session.get_credentials.access_key.return_value = \
+            sentinel.access_key
+        mock_Session.get_credentials.secret_key.return_value = \
+            sentinel.secret_key
+        mock_Session.get_credentials.method.return_value = \
+            sentinel.method
```

Comment:
Not sure mock_Session.get_credentials return values need to be added. I can't see where they are used later on in this test...